staging/src/k8s.io/kube-scheduler/config/v1beta1/types_pluginargs.go

```diff
@@ -94,6 +94,10 @@ type NodeResourcesFitArgs struct {
 	// A resource group name can't contain '/'.
 	// +listType=atomic
 	IgnoredResourceGroups []string `json:"ignoredResourceGroups,omitempty"`
+
+	// ScoringStrategy selects the node resource scoring strategy.
+	// +listType=atomic
+	ScoringStrategy ScoringStrategy `json:"scoringStrategy,omitempty"`
```

Comment:
ah, there is a misunderstanding, this needs to be a reference, only the internal type should be non-reference

Reply:
So config.ScoringStrategy is 
```go
ScoringStrategy ScoringStrategy
```
but  v1beta1.ScoringStrategy and v1beta2.ScoringStrategy is 
```go
ScoringStrategy *ScoringStrategy
```

Reply:
Yes.

Reply:
> Yes.

That's will result in like this 
```go
WARNING: in.ScoringStrategy requires manual conversion: inconvertible types (k8s.io/kubernetes/pkg/scheduler/apis/config.ScoringStrategy vs *k8s.io/kube-scheduler/config/v1beta1.ScoringStrategy)
```

Reply:
I did `make update` doesn't resolve this error, @ahg-g how to handle this?

Reply:
We need a manual conversion function, let me handle it in a followup PR.

Please revert the change you made last, and lets just focus on validation for now: https://github.com/kubernetes/kubernetes/pull/101822#issuecomment-863999586

1) in v1beta1, the new scoring plugin can't be used with LeastAllocated/MostAllocated/RequestedToCapacityRatio plugins. Add the logic to the end of this function https://github.com/kubernetes/kubernetes/blob/b5243341053cf9f67b1a163ccfea07517755b7a8/pkg/scheduler/apis/config/validation/validation.go#L120, probably rename it validatePluginSet . Basically iterate over the set and return an error if both `NodeResourcesFit` and any of LeastAllocated/MostAllocated/RequestedToCapacityRatio exist in the same set.

2) in v1beta2, deprecate LeastAllocated, MostAllocated and RequestedToCapacityRatio plugins: basically add them here: https://github.com/kubernetes/kubernetes/blob/b5243341053cf9f67b1a163ccfea07517755b7a8/pkg/scheduler/apis/config/validation/validation.go#L100

Reply:
> We need a manual conversion function, let me handle it in a followup PR.
> 
> 
> 
> Please revert the change you made last, and lets just focus on validation for now: https://github.com/kubernetes/kubernetes/pull/101822#issuecomment-863999586
> 
> 
> 
> 1) in v1beta1, the new scoring plugin can't be used with LeastAllocated/MostAllocated/RequestedToCapacityRatio plugins. Add the logic to the end of this function https://github.com/kubernetes/kubernetes/blob/b5243341053cf9f67b1a163ccfea07517755b7a8/pkg/scheduler/apis/config/validation/validation.go#L120, probably rename it validatePluginSet . Basically iterate over the set and return an error if both `NodeResourcesFit` and any of LeastAllocated/MostAllocated/RequestedToCapacityRatio exist in the same set.
> 
> 
> 
> 2) in v1beta2, deprecate LeastAllocated, MostAllocated and RequestedToCapacityRatio plugins: basically add them here: https://github.com/kubernetes/kubernetes/blob/b5243341053cf9f67b1a163ccfea07517755b7a8/pkg/scheduler/apis/config/validation/validation.go#L100
> 
> 

yes，i nearly finish this validation, after finish revert,the commit will be ready.